### PR TITLE
mark `stop`, `skip`, `@epochs` as deprecated

### PR DESCRIPTION
**Pull request by mcabbott**
Replaces #1913 with the simpler plan to remove these from Flux 0.14.

Closes #1913

### CODE REVIEW

**Comment by Saransh-cpp** [comment]
Should the occurrences of `stop`, `skip`, and `@epochs` be removed from the docs, or should this wait till they are removed (`v0.14`)? The doctests are failing due to the deprecation warnings. Should the doctests be removed, or should they be updated with the warnings?

**Comment by mcabbott** [comment]
That's a good question. I think that:
* For the dep-warns to be of any use, I guess this must be merged into a 0.13 release.  
* The removal of both the code and the docs should be in 0.14, thus a different PR
* Possibly the last 0.13 docs should have a warning note added too? That could be this PR.
* To keep tests passing, maybe this PR should also remove some `jldoctest` annotations.

**Comment by mcabbott** [comment]
@Saransh-cpp maybe you know, are the built docs previewed somewhere? Is this `!!! note` thing going to work?

**Comment by Saransh-cpp** [comment]
@mcabbott the previews are only generated if a PR is created from a branch belonging to the root `Flux` repository (restrictions from `Documenter.jl`). If the PR is from a branch belonging to the root repository, adding the `documentation` tag to the PR will trigger the `DocBot` to comment the generated URL.

**Comment by Saransh-cpp** [comment]
`!!! note` works! Tried this locally!

**Comment by mcabbott** [comment]
Ready to merge?

File: src/optimise/train.jl [hunk 1]
```diff
@@ -46,6 +49,9 @@ end
 ```
 """
 function skip()
+  Base.depwarn("""Flux.skip() will be removed from Flux 0.14.
+                  It never worked as described,
```
Comment (ToucheSir) [inline src/optimise/train.jl:53]:
```suggestion
```

File: src/optimise/train.jl [hunk 2]
```diff
@@ -150,6 +164,8 @@ hello
 ```
 """
 macro epochs(n, ex)
+  Base.depwarn("""The macro `@epochs` will be removed from Flux 0.14.
+                  Just write an ordinary for loop.""", Symbol("@epochs"), force=true)
```
Comment (ToucheSir) [inline src/optimise/train.jl:168]:
```suggestion
                  As an alternative, you can write a simple `for i in 1:epochs` loop.""", Symbol("@epochs"), force=true)
```

**Review by ToucheSir — COMMENTED** [review]
A bit of wordsmithing, but otherwise good to go

**Review by ToucheSir — APPROVED** [review]

**Pull request merged**